### DISCUSSION TITLE
Can pyscript access the /dev/ttyUSBx ports?

### DISCUSSION THREAD

**Discussion by JayAr-ZA** [custom-components/pyscript#256] [Q&A]
Hello, I have an WaveShare RS485 / Modbus USB dongle, that drives automation of the likes of Relays using the WaveShare 8CH relay, and I want to also connect to modbus Temperature meters, etc.

The code to interact with the modbus devices, through the USB dongle is in Python3. 

Q: Can pyscript access the /dev/ttyUSB0 or others, using normal python libraries like Serial etc or is it running in a contained env that cannot get to physical ports etc?

**Comment by dlashua** [comment]
yes it can.
